### PR TITLE
move out LinearAlgebra into its own repository

### PR DESCRIPTION
**Pull request by KristofferC**
This moves out LinearAlgebra into its own repo https://github.com/JuliaLang/LinearAlgebra.jl. This repo is still a bit bare (README needs to be added) but it has CI set up to run on buildkite (https://buildkite.com/julialang/linearalgebra-dot-jl/builds/18) and doc building on GHA. The external repo has all commits up to https://github.com/JuliaLang/julia/commit/4709b6c48e79f6226e6dbee1b49bf7e563058ff7 included in it.

The reason for the move is to be able to focus issues and PRs and development regarding LinearAlgebra in one place.

### CODE REVIEW

File: julia.spdx.json [hunk 1]
```diff
@@ -86,6 +86,18 @@
             "copyrightText": "Copyright (c) 2020 Stefan Karpinski <stefan@karpinski.org> and contributors",
             "summary": "ArgTools provides tools for creating consistent, flexible APIs that work with various kinds of function arguments."
         },
+        {
+            "name": "LinearAlgebra.jl",
+            "SPDXID": "SPDXRef-JuliaLinearAlgebra",
+            "downloadLocation": "git+https://github.com/Julia/LinearAlgera.jl.git",
```
Comment (mofeing) [inline julia.spdx.json:92]: there are some typos in here
```suggestion
            "downloadLocation": "git+https://github.com/JuliaLang/LinearAlgebra.jl.git",
```

**Comment by IanButterworth** [comment]
Screenshot 2024-11-22 at 9 18 46 AM
Shame github has stopped showing lines removed on https://github.com/JuliaLang/julia/graphs/contributors

**Comment by KristofferC** [comment]
Heyy, when did they remove that :(

**Comment by IanButterworth** [comment]
I heard about it on the Julia Dispatch Podcast #ad

**Comment by gitboy16** [comment]
Hi, might be a dumb question but I was wondering if there is a way to build Julia without LinearAlgebra? More generally speaking is there a mechanism to choose which library to include or exclude while building Julia?

**Comment by giordano** [comment]
> Heyy, when did they remove that :(

I'm pretty sure it was since one year ago, they announced it on the blog, can't find the post now though.

**Comment by ViralBShah** [comment]
Should we turn off LinearAlgebra testing on the Julia repo now for every commit?

**Comment by IanButterworth** [comment]
Could we run it as a regular GitHub action test on Linux only, like we do for Revise? That way it's in parallel so should still have the same effect of speeding up CI, but less chance of regression.

**Comment by ViralBShah** [comment]
Even so - the full LinearAlgebra testsuite is overkill - but that is a separate issue being discussed in https://github.com/JuliaLang/LinearAlgebra.jl/issues/1117.

**Comment by KristofferC** [comment]
I suggested running it when:

- The `LinearAlgebra.version` file is changed.
- Om backports and release branches.

**Comment by DilumAluthge** [comment]
Here's a tracking issue: https://github.com/JuliaCI/julia-buildkite/issues/411

**Comment by waldyrious** [comment]
Just for historical reference: this PR closed issue [#22698](https://github.com/JuliaLang/julia/issues/22698), "Move LinearAlgebra to its own repo" (which now lives at https://github.com/JuliaLang/LinearAlgebra.jl/issues/446).

**Pull request merged**